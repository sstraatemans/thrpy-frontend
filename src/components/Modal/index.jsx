// @flow
import React, { Component } from "react";
import {
  StyledModal,
  ModalContent,
  ModalFooter,
  ModalTitle,
  ModalWrapper,
  ModalOverlay
} from "./style";
import Button, { ButtonTypes } from "./../Button";

type Props = {
  hasFixedFooter?: boolean,
  onDismiss: Function,
  onOk: Function,
  children: Array<any>
};

type State = {
  isClosing: boolean
};

class Modal extends Component<Props, State> {
  handleDismiss: Function;
  closeModal: Function;
  modal: any;
  constructor(props: Props) {
    super(props);

    this.state = {
      isClosing: false
    };
    this.handleDismiss = this.handleDismiss.bind(this);
    this.closeModal = this.closeModal.bind(this);
  }
  handleDismiss() {
    this.setState({ isClosing: true });
    this.modal.addEventListener("animationend", this.closeModal);
  }

  closeModal() {
    this.props.onDismiss();
  }

  componentDidMount() {
    console.log("Did Mount");
  }
  componentWillUnmount() {
    console.log("gone?");
  }

  render() {
    return (
      <ModalWrapper>
        <ModalOverlay isClosing={this.state.isClosing} isOpen="true" />
        <StyledModal
          innerRef={comp => {
            this.modal = comp;
          }}
          isClosing={this.state.isClosing}
          isOpen="true"
          hasFixedFooter={this.props.hasFixedFooter}
        >
          <ModalContent hasFixedFooter={this.props.hasFixedFooter}>
            <ModalTitle>This is a title</ModalTitle>
            {this.props.children}
          </ModalContent>
          <ModalFooter hasFixedFooter={this.props.hasFixedFooter}>
            <Button onClick={this.handleDismiss} style={ButtonTypes.Flat}>
              Cancel
            </Button>
            <Button onClick={this.props.onOk} style={ButtonTypes.Flat}>
              Ok
            </Button>
          </ModalFooter>
        </StyledModal>
      </ModalWrapper>
    );
  }
}

export default Modal;
