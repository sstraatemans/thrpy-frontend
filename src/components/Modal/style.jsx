import styled, { keyframes } from "styled-components";
import { H4 } from "./../../style";

export const ModalWrapper = styled.div``;
export const ModalOverlay = styled.div`
  position: fixed;
  display: block;
  border-radius: 50%;
  align-items: center;
  justify-content: center;
  z-index: 998;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  width: 5vw;
  height: 5vw;
  background: #000;
  opacity: 0.5;

  animation-duration: 0.5s;
  animation-delay: 0s;
  animation-fill-mode: forwards;
  animation-play-state: running;
  transform: translate(50vw, 50vh) scale(1);

  ${props =>
    props.isOpen &&
    !props.isClosing &&
    `
    animation-name: ${openOverlayAnimation};
    animation-direction: normal;
    display: block;
  `};

  ${props =>
    props.isClosing &&
    `
    transform: translate(50vw, 50vh) scale(40);
    animation-name: ${openOverlayAnimationReverse};
    animation-duration: 0.5s;
    animation-delay: 0.3s;
    animation-fill-mode: forwards;
    animation-play-state: running;
    display: block;
  `};
`;
export const StyledModal = styled.div`
  display: none;
  position: fixed;
  opacity: 1;
  left: 0;
  right: 0;
  background-color: #fafafa;
  padding: 0;
  max-height: 70%;
  width: 55%;
  margin: auto;
  overflow-y: auto;
  border-radius: 2px;
  z-index: 1000;
  transform-origin: center;

  box-shadow: ${props => props.theme.boxShadow};
  transform: scale(0.7, 1.2);
  animation-duration: 0.5s;
  animation-delay: 0.2s;
  animation-fill-mode: forwards;
  animation-play-state: running;

  ${props =>
    props.isOpen &&
    !props.isClosing &&
    `
    opacity: 0;
    animation-name: ${openModalAnimation};
    animation-direction: normal;
    display: block;
  `};

  ${props =>
    props.isClosing &&
    `
    animation-name: ${openModalAnimationReverse};
    animation-duration: 0.5s;
    animation-delay: 0s;
    animation-fill-mode: forwards;
    animation-play-state: running;
    display: block;
  `};
`;

export const ModalContent = styled.div`padding: 24px;`;

export const ModalFooter = styled.div`
  border-radius: 0 0 2px 2px;
  background-color: #fafafa;
  padding: 4px 6px;
  height: 56px;
  width: 100%;
  text-align: right;

  a {
    margin: 6px 0;
  }
`;

export const ModalTitle = H4.extend``;

const openModalAnimation = keyframes`
  0% {
  }
  1% {
    display: bock;
    opacity: .5;
    transform: scale(0.7, 1.2);
  }
  100% {
    opacity: 1;
    transform: scale(1, 1);
  }
`;

const openModalAnimationReverse = keyframes`
  0% {
    opacity: 1;
    transform: scale(1, 1);
  }
  99%{
    opacity: .5;
    transform: scale(0.7, 1.2);
  }
  100% {
    opacity: 0;
  }
`;

const openOverlayAnimation = keyframes`
  0% {
    opacity: 0;
    transform: translate(50vw, 50vh) scale(1);
  }
  100% {
    opacity: .5;
    transform: translate(50vw, 50vh) scale(40);
  }
`;

const openOverlayAnimationReverse = keyframes`
0% {
  transform: translate(50vw, 50vh) scale(40);
}
  1% {
    opacity: .5;
    transform: translate(50vw, 50vh) scale(40);
  }
  100% {
    opacity: 0;
    transform: translate(50vw, 50vh) scale(.2);

  }
`;
