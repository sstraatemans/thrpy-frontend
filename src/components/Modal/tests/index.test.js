import React from "react";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Modal from "./../";

describe("<Modal />", () => {
  it("render component <Modal>", () => {
    const wrapper = mount(<Modal />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
