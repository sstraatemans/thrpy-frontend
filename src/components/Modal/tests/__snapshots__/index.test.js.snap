exports[`<Modal /> render component <Modal> 1`] = `
<Modal>
  <styled.div>
    <div
      className="sc-htpNat ktprKK">
      <styled.div>
        <div
          className="sc-bxivhb iuNRCJ">
          <styled.h4>
            <h4
              className="sc-EHOje flMdcs">
              This is a title
            </h4>
          </styled.h4>
          his is text
        </div>
      </styled.div>
      <styled.div>
        <div
          className="sc-ifAKCX jlvqiv">
          <Button
            onClick={undefined}
            style="Flat">
            <styled.a
              onClick={undefined}>
              <a
                className="sc-bZQynM jwYwAf"
                onClick={undefined}>
                Cancel
              </a>
            </styled.a>
          </Button>
          <Button
            onClick={undefined}
            style="Flat">
            <styled.a
              onClick={undefined}>
              <a
                className="sc-bZQynM jwYwAf"
                onClick={undefined}>
                Ok
              </a>
            </styled.a>
          </Button>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Modal>
`;
